simples3.go

```diff
@@ -52,6 +52,7 @@ type UploadInput struct {
 	// optional fields
 	ContentDisposition string
 	ACL                string
+	Metadata           map[string]string
```

Comment:
https://docs.aws.amazon.com/AmazonS3/latest/userguide/UsingMetadata.html

Just went through the doc, I think this might be confusing if we name it `Metadata`

Let's rename to `CustomMetadata` and add a go doc comment

```
CustomMetadata map[string]string // Setting key/value pairs adds user-defined metadata keys to the object, prefixed with AMZMetaPrefix
```